vtex-apps/tiktok-tbp
# Feature/brpa 712 handle functionalities i ds

#### What is the purpose of this pull request?

- Get business profile data and save it on app settings.
  - Before retrieve the data, the passphrase is checked. In case of invalid passphrase or any error during the workflow, the status `error` is saved in the app settings and an error is returned. If everything is OK, success is returned and status `connected` is saved in the app settings.
- Add a mapper to map the data returned by TikTok.
  - The data returned from TikTok API is in the `snake_case` pattern. This mapper turns them into `camelCase` pattern.

#### How to test it?

You can see the data saved in the app settings [here](https://tiktoktbp--sandboxbrdev.myvtex.com/admin/apps/vtex.tiktok-tbp@0.3.0/setup/).

#### Screenshots or example usage

![image](https://user-images.githubusercontent.com/50276597/154129876-844eae54-1590-4a66-9be8-b57590919095.png)

#### How does this PR make you feel? [:link:](http://giphy.com/)



![](https://media.giphy.com/media/j1gDGxFWrAaRbzbMx5/giphy-downsized.gif)

#### Types of changes
